core/src/com/po/fuck/model/weapons/LaserBeam.java

```diff
@@ -1,49 +1,51 @@
-package com.po.fuck.weapons;
+package com.po.fuck.model.weapons;
 
 import java.util.List;
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.graphics.g2d.Animation;
 import com.badlogic.gdx.graphics.g2d.Sprite;
-import com.badlogic.gdx.graphics.g2d.TextureRegion;
 import com.badlogic.gdx.math.Polygon;
 import com.badlogic.gdx.math.Vector2;
-import com.po.fuck.Entity;
-import com.po.fuck.GeometryMisc;
-import com.po.fuck.GifDecoder;
-import com.po.fuck.collision.Collidable;
-import com.po.fuck.collections.All;
-import com.po.fuck.lifetime.Manager;
+import com.po.fuck.model.collision.Collidable;
+import com.po.fuck.model.Entity;
+import com.po.fuck.model.GeometryMisc;
+import com.po.fuck.model.lifetime.Manager;
+import com.po.fuck.view.GifDecoder;
+import com.po.fuck.model.collections.All;
+
+import static com.po.fuck.model.Constants.LASER_BEAM_DAMAGE;
+import static com.po.fuck.model.Constants.LASER_BEAM_LIFE_TIME;
 
 public final class LaserBeam extends Bullet {
     // version is also good, but for some reason it renders with artifacts
-    static private final Animation<TextureRegion> animation = GifDecoder.loadGIFAnimation(Animation.PlayMode.LOOP, Gdx.files.internal("laser3.gif").read());
-    private final float LIVE_TIME = 1;
-    private float timeElapsed = 0;
 
     {
-        damage = 3;
-        this.sprite = new Sprite(animation.getKeyFrame(0));
+        damage = LASER_BEAM_DAMAGE;
+        lifeTime = LASER_BEAM_LIFE_TIME;
+        /* Temp solution, afterwards use AssetManager or .properties later */
+        width = new Sprite(GifDecoder.loadGIFAnimation(Animation.PlayMode.LOOP, Gdx.files.internal("laser3.gif").read())
+                                .getKeyFrame(0)).getWidth();
+        height = new Sprite(GifDecoder.loadGIFAnimation(Animation.PlayMode.LOOP, Gdx.files.internal("laser3.gif").read())
+                                .getKeyFrame(0)).getHeight();
     }
 
     LaserBeam(Vector2 muzzle_position, Vector2 direction, int teamTag) {
         this.teamTag = teamTag;
-        this.position = muzzle_position.cpy().add(direction.cpy().setLength(sprite.getWidth() / 2));
+        this.position = muzzle_position.cpy().add(direction.cpy().setLength(width / 2));
         this.velocity = direction.cpy().setLength(0.1f);
     }
 
     @Override
     public void update(float delta) {
         timeElapsed += delta;
-        this.sprite = new Sprite(animation.getKeyFrame(timeElapsed));
-        sprite.setRotation(-velocity.angleDeg());
 
-        if (timeElapsed > LIVE_TIME) {
+        if (timeElapsed > lifeTime) {
             Manager.destroy_raw(this);
             return;
         }
 
-        Polygon polygon = GeometryMisc.createRectangle(position, sprite);
+        Polygon polygon = GeometryMisc.createRectangle(position, width, height, -velocity.angleDeg());
```

Comment:
Why are you passing `-velocity.angleDeg()` and then doing negation (* -1) inside of this function ?